src/Pages/SignUpForm.jsx

```diff
@@ -0,0 +1,284 @@
+      <Container className="signup-container">
+        <Card className="signup-card">
+          <h2 className="text-center mb-1">Sign Up</h2>
+
+          <Form noValidate onSubmit={handleSubmit}>
+            <Form.Group className="mb-1 px-4">
+              <Form.Label>First Name</Form.Label>
+              <Form.Control
+                type="text"
+                name="firstName"
+                value={formData.firstName}
+                onChange={handleValueChange}
+                isInvalid={!!formErrors.firstName}
+                isValid={formErrors.firstName === ''}
+              />
+              <Form.Control.Feedback type="invalid">
+                {formErrors.firstName}
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-1 px-4">
+              <Form.Label>Last Name</Form.Label>
+              <Form.Control
+                type="text"
+                name="lastName"
+                value={formData.lastName}
+                onChange={handleValueChange}
+                isInvalid={!!formErrors.lastName}
+                isValid={formErrors.lastName === ''}
+              />
+              <Form.Control.Feedback type="invalid">
+                {formErrors.lastName}
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-1 px-4">
+              <Form.Label>Email</Form.Label>
+              <Form.Control
+                type="email"
+                name="email"
+                value={formData.email}
+                onChange={handleValueChange}
+                isInvalid={!!formErrors.email}
+                isValid={formErrors.email === ''}
+              />
+              <Form.Control.Feedback type="invalid">
+                {formErrors.email}
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-1 px-4">
+              <Form.Label>Role</Form.Label>
+              <Form.Control
+                as="select"
+                name="roleId"
+                value={formData.roleId}
+                onChange={handleValueChange}
+                isInvalid={!!formErrors.roleId}
+                isValid={formErrors.roleId === ''}
+              >
+                <option value="">Select Role</option>
+                {rolesData.length > 0 ? (
+                  rolesData.map((role) => (
+                    <option key={role.id} value={role.id}>
+                      {role.name}
+                    </option>
+                  ))
+                ) : (
+                  <option disabled>No roles available</option>
+                )}
+              </Form.Control>
+              <Form.Control.Feedback type="invalid">
+                {formErrors.roleId}
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group controlId="formPassword" className="mb-1 px-4">
+              <Form.Label>Password</Form.Label>
```

Comment:
create a separate component for password field and use from there.
Its not looking nice